app/controllers/submit_source_controller.rb

```diff
@@ -0,0 +1,16 @@
+class SubmitSourceController < ApplicationController
+  def new
+    @submit_source = SubmitSource.new
+  end
+
+  def create
+    @submit_source = SubmitSource.new(params[:submit_source])
+    @submit_source.request = request
+    if @submit_source.deliver
+      flash.now[:error] = nil
+    else
+      flash.now[:error] = 'Cannot submit source.'
```

Comment:
Prefer double-quoted strings unless you need single quotes to avoid extra backslashes for escaping.